src/main/java/com/twentyonec/ItemsLogger/utils/Storage.java

```diff
@@ -159,44 +151,47 @@ public void setUpTable() {
 		this.update(createTableQuery);
 	}
 
-	private ItemPlayer initializePlayer(ResultSet rs) {
+	private ItemPlayer initializePlayer(final ResultSet rs) {
 
 		try {
-			UUID uuid = UUID.fromString(rs.getString("uuid"));
-			String inv = rs.getString("inventory");
-			String cause = rs.getString("cause");
-			int x = rs.getInt("loc_x");
-			int y = rs.getInt("loc_y");
-			int z = rs.getInt("loc_z");
-			int experience = rs.getInt("experience");
-			Date date = rs.getDate("date");
-			Time time = rs.getTime("time");
+			final UUID uuid = UUID.fromString(rs.getString("uuid"));
+			final String inv = rs.getString("inventory");
+			final String cause = rs.getString("cause");
+			final int x = rs.getInt("loc_x");
+			final int y = rs.getInt("loc_y");
+			final int z = rs.getInt("loc_z");
+			final int experience = rs.getInt("experience");
+			final Date date = rs.getDate("date");
+			final Time time = rs.getTime("time");
 
 			return new ItemPlayer(uuid, inv, cause, x, y, z, experience, date, time);
-		} catch (SQLException e) {
+		} catch (final SQLException e) {
 			e.printStackTrace();
 			return null;
 		}
 	}
 
-	private CompletableFuture<ItemPlayer[]> retrieveLogListAsync(UUID uuid, String date, String cause, int max, int min) {
+	private CompletableFuture<ItemPlayer[]> retrieveLogListAsync(final UUID uuid, final String date, final String cause, final int max, final int min) {
 
 		return CompletableFuture.supplyAsync(() -> {
-			String sql = " SELECT * FROM itemslogger WHERE uuid = '" + uuid + "'";
-
-			if (cause != null) {
-				sql += " AND cause = '" + cause + "'";
-			}
+			final String sql = " SELECT * FROM itemslogger WHERE uuid = ?";
 
+			final StringBuilder sb = new StringBuilder(sql);
+			final List<Object> args = new ArrayList<Object>();
+			args.add(uuid.toString());
 			if (date != null) {
-				sql += " AND date = '" + date + "'";
+				sb.append(" AND date = ?");
+				args.add(date);
 			}
-
-			sql += " ORDER BY date DESC, time DESC";
-			ResultSet rs = storage.query(sql);
+			if (cause != null) {
+				sb.append(" AND cause = ?");
+				args.add(cause);
+			}
+			sb.append(" ORDER BY date DESC, time DESC");
+			final ResultSet rs = storage.query(sb.toString(), args.toArray(new Object[0]));
 
 			try {
-				List<ItemPlayer>playerDataArray = new ArrayList<ItemPlayer>();
+				final List<ItemPlayer>playerDataArray = new ArrayList<ItemPlayer>();
 				int i = 0;
 				while (rs.next()) {
```

Comment:
Add `Limit ?, ?` to your query and use `min` and `max - min` as the parameters.
The first parameter is how many rows to skip and the second is how many to return from that point.
This process is called paging and is pretty important to learn for database stuff. Right now you're pulling all the rows associated with a user in on every query and then throwing away a bunch of them; it's very wasteful
https://dev.mysql.com/doc/refman/8.0/en/limit-optimization.html